### PR TITLE
NamedType AST

### PR DESCRIPTION
**Pull request by leebyron**
This replaces the raw usage of `Name` as a kind of `Type` with the AST node `NamedType`. It also carries the vernacular "named type" throughout the codebase in the places where this same concept was being provided many different names (unmodified, naked, raw, etc).

Finally, this simplifies some of the validation rules once this clarification was in place.

### CODE REVIEW

File: src/validator/rules/FragmentsOnCompositeTypes.js [hunk 1]
```diff
@@ -29,21 +30,22 @@ export default function FragmentsOnCompositeType(
...
-      var type = context.getSchema().getType(typeName);
-      if (!isCompositeType(type)) {
+      var type = context.getType();
+      if (type && !isCompositeType(type)) {
         return new GraphQLError(
-          fragmentOnNonCompositeErrorMessage(node.name.value, typeName),
+          fragmentOnNonCompositeErrorMessage(
+            node.name.value,
+            print(node.typeCondition)
+          ),
```
Comment (schrockn) [inline src/validator/rules/FragmentsOnCompositeTypes.js]: I assume all these changes are unrelated?

Comment (leebyron) [inline src/validator/rules/FragmentsOnCompositeTypes.js]: Related. `typeCondition` changed from `Name` to `NamedType` which necessitated changing how it's printed.

Comment (schrockn) [inline src/validator/rules/FragmentsOnCompositeTypes.js]: ah cool

**Comment by schrockn** [comment]
push this bad boy. lgtm.
